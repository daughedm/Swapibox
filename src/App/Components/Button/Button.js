import React from 'react';
import Vader from '../../../Assets/Images/vader.svg';
import DeathStar from '../../../Assets/Images/deathstar.svg';
import Falcon from '../../../Assets/Images/falcon.svg';
import '../Button/Button.css';

const Button = ({ handleClick, category, favorites }) => {

  return (
    <div className="buttons-wrapper" >
      <div 
        className="button-container" 
        onClick={() => handleClick('people')} 
        type="button" 
      >
        <div className="circle">
          <img 
            src={Vader} 
            alt="people" 
            className="icon"/>
        </div>
        <h2 className="btnTitle">PEOPLE</h2>
      </div>
      <div 
        className="button-container" 
        onClick={() => handleClick('planets')} 
        type="button">
        <div className="circle">
          <img 
            src={DeathStar} 
            alt="planets" 
            className="icon"/>
        </div>
        <h2 className="btnTitle">PLANETS</h2>
      </div>
      <div 
        className="button-container" 
        onClick={() => handleClick('vehicles')} 
        type="button">
        <div className="circle" >
          <img 
            src={Falcon} 
            alt="vehicles" 
            className="icon"/>
        </div>
        <h2 className="btnTitle">VEHICLES</h2>
      </div>
      <div className="button-container divider">
        <div className="circle filled">
          <h3 className="favorite-number">{favorites.length ? favorites.length : 0}</h3>
        </div>
        <h2 className="btnTitle">FAVORITES</h2>
      </div>
    </div>
  )
}

export default Button;