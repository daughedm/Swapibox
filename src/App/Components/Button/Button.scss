$sw-yellow: rgb(246, 231, 48);
$light-bg: rgb(214, 215, 231);

.favorite-number {
  text-align: center;
  margin: 14px;
  font-size: 2.2rem;
}

.buttons-wrapper {
  display: flex;
  align-content: space-between;
  padding-right: 60px;
}

.button-container {
  padding: 35px 15px 0 15px;
} 

.circle {
  border-radius: 100%;
  border: 3px solid $sw-yellow;
  // background-color: $light-bg;
  width: 70px;
  height: 70px;
}

.circle:hover {
  border: 3px dashed $sw-yellow;
  cursor: pointer;
}

.border {
  border: 3px dashed $sw-yellow;
}

.icon {
  width: 70%;
  display: block;
  margin: auto;
  padding-top: 10px;
}

.btnTitle {
  color: $light-bg;
  display: block;
  font-size: 0.8rem;
  text-align: center;
  margin-top: 10px;
}

.btnTitle:hover {
  color: $sw-yellow;
  padding-bottom: 5px;
  border-bottom: 2px solid $sw-yellow;
}

.filled {
  background-color: transparent;
  color: $sw-yellow;
  text-align: center;
}

.divider {
  border-left: 2px solid $light-bg;
  margin-left: 25px;
  padding-left: 40px;
}