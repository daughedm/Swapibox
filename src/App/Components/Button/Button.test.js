import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  let wrapper;

  it('should match snapshot', () => {
    const favorites = ['favorite'];
    const handleClick = jest.fn();
    
    wrapper = shallow(<Button 
      category="people"
      handleClick={handleClick} 
      favorites={favorites}/>)

    expect(wrapper).toMatchSnapshot();
  });
})
