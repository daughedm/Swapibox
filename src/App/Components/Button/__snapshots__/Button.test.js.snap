// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    category="people"
    favorites={
        Array [
            "favorite",
          ]
    }
    handleClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="people-button"
          onClick={[Function]}
          type="button"
>
          <div
                    className="circle"
          >
                    <img
                              alt="people"
                              className="icon"
                              src="vader.svg"
                    />
          </div>
          <h2
                    className="btnTitle"
          >
                    PEOPLE
          </h2>
</div>,
        <div
          className="button-container"
          id="planet-button"
          onClick={[Function]}
          type="button"
>
          <div
                    className="circle"
          >
                    <img
                              alt="planets"
                              className="icon"
                              src="deathstar.svg"
                    />
          </div>
          <h2
                    className="btnTitle"
          >
                    PLANETS
          </h2>
</div>,
        <div
          className="button-container"
          id="vehicle-button"
          onClick={[Function]}
          type="button"
>
          <div
                    className="circle"
          >
                    <img
                              alt="vehicles"
                              className="icon"
                              src="falcon.svg"
                    />
          </div>
          <h2
                    className="btnTitle"
          >
                    VEHICLES
          </h2>
</div>,
        <div
          className="button-container divider"
          id="favorite-button"
          onClick={[Function]}
>
          <div
                    className="circle filled"
          >
                    <h3
                              className="favorite-number"
                    >
                              1
                    </h3>
          </div>
          <h2
                    className="btnTitle"
          >
                    FAVORITES
          </h2>
</div>,
      ],
      "className": "buttons-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="circle"
>
              <img
                            alt="people"
                            className="icon"
                            src="vader.svg"
              />
</div>,
            <h2
              className="btnTitle"
>
              PEOPLE
</h2>,
          ],
          "id": "people-button",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="people"
                className="icon"
                src="vader.svg"
/>,
              "className": "circle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "people",
                "className": "icon",
                "src": "vader.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PEOPLE",
              "className": "btnTitle",
            },
            "ref": null,
            "rendered": "PEOPLE",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="circle"
>
              <img
                            alt="planets"
                            className="icon"
                            src="deathstar.svg"
              />
</div>,
            <h2
              className="btnTitle"
>
              PLANETS
</h2>,
          ],
          "className": "button-container",
          "id": "planet-button",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="planets"
                className="icon"
                src="deathstar.svg"
/>,
              "className": "circle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "planets",
                "className": "icon",
                "src": "deathstar.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PLANETS",
              "className": "btnTitle",
            },
            "ref": null,
            "rendered": "PLANETS",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="circle"
>
              <img
                            alt="vehicles"
                            className="icon"
                            src="falcon.svg"
              />
</div>,
            <h2
              className="btnTitle"
>
              VEHICLES
</h2>,
          ],
          "className": "button-container",
          "id": "vehicle-button",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="vehicles"
                className="icon"
                src="falcon.svg"
/>,
              "className": "circle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "vehicles",
                "className": "icon",
                "src": "falcon.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "VEHICLES",
              "className": "btnTitle",
            },
            "ref": null,
            "rendered": "VEHICLES",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="circle filled"
>
              <h3
                            className="favorite-number"
              >
                            1
              </h3>
</div>,
            <h2
              className="btnTitle"
>
              FAVORITES
</h2>,
          ],
          "className": "button-container divider",
          "id": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3
                className="favorite-number"
>
                1
</h3>,
              "className": "circle filled",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "favorite-number",
              },
              "ref": null,
              "rendered": 1,
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "FAVORITES",
              "className": "btnTitle",
            },
            "ref": null,
            "rendered": "FAVORITES",
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="people-button"
            onClick={[Function]}
            type="button"
>
            <div
                        className="circle"
            >
                        <img
                                    alt="people"
                                    className="icon"
                                    src="vader.svg"
                        />
            </div>
            <h2
                        className="btnTitle"
            >
                        PEOPLE
            </h2>
</div>,
          <div
            className="button-container"
            id="planet-button"
            onClick={[Function]}
            type="button"
>
            <div
                        className="circle"
            >
                        <img
                                    alt="planets"
                                    className="icon"
                                    src="deathstar.svg"
                        />
            </div>
            <h2
                        className="btnTitle"
            >
                        PLANETS
            </h2>
</div>,
          <div
            className="button-container"
            id="vehicle-button"
            onClick={[Function]}
            type="button"
>
            <div
                        className="circle"
            >
                        <img
                                    alt="vehicles"
                                    className="icon"
                                    src="falcon.svg"
                        />
            </div>
            <h2
                        className="btnTitle"
            >
                        VEHICLES
            </h2>
</div>,
          <div
            className="button-container divider"
            id="favorite-button"
            onClick={[Function]}
>
            <div
                        className="circle filled"
            >
                        <h3
                                    className="favorite-number"
                        >
                                    1
                        </h3>
            </div>
            <h2
                        className="btnTitle"
            >
                        FAVORITES
            </h2>
</div>,
        ],
        "className": "buttons-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="circle"
>
                <img
                                alt="people"
                                className="icon"
                                src="vader.svg"
                />
</div>,
              <h2
                className="btnTitle"
>
                PEOPLE
</h2>,
            ],
            "id": "people-button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="people"
                  className="icon"
                  src="vader.svg"
/>,
                "className": "circle",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "people",
                  "className": "icon",
                  "src": "vader.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PEOPLE",
                "className": "btnTitle",
              },
              "ref": null,
              "rendered": "PEOPLE",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="circle"
>
                <img
                                alt="planets"
                                className="icon"
                                src="deathstar.svg"
                />
</div>,
              <h2
                className="btnTitle"
>
                PLANETS
</h2>,
            ],
            "className": "button-container",
            "id": "planet-button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="planets"
                  className="icon"
                  src="deathstar.svg"
/>,
                "className": "circle",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "planets",
                  "className": "icon",
                  "src": "deathstar.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PLANETS",
                "className": "btnTitle",
              },
              "ref": null,
              "rendered": "PLANETS",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="circle"
>
                <img
                                alt="vehicles"
                                className="icon"
                                src="falcon.svg"
                />
</div>,
              <h2
                className="btnTitle"
>
                VEHICLES
</h2>,
            ],
            "className": "button-container",
            "id": "vehicle-button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="vehicles"
                  className="icon"
                  src="falcon.svg"
/>,
                "className": "circle",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "vehicles",
                  "className": "icon",
                  "src": "falcon.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "VEHICLES",
                "className": "btnTitle",
              },
              "ref": null,
              "rendered": "VEHICLES",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="circle filled"
>
                <h3
                                className="favorite-number"
                >
                                1
                </h3>
</div>,
              <h2
                className="btnTitle"
>
                FAVORITES
</h2>,
            ],
            "className": "button-container divider",
            "id": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="favorite-number"
>
                  1
</h3>,
                "className": "circle filled",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "className": "favorite-number",
                },
                "ref": null,
                "rendered": 1,
                "type": "h3",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "FAVORITES",
                "className": "btnTitle",
              },
              "ref": null,
              "rendered": "FAVORITES",
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
